sdk/src/main/scala/com/horizen/account/receipt/EthereumReceipt.scala

```diff
@@ -11,70 +11,85 @@ import java.util
 
-    if (contractAddress != null)
-      contractAddressStr = BytesUtils.toHexString(contractAddress)
+    contractAddressStr = contractAddress match {
+      case Some(addr) => BytesUtils.toHexString(addr)
+      case None => null
+    }
 
-    val infoNonConsensusStr : String =
-      String.format(s" - (receipt non consensus data) {txHash=$txHashStr, txIndex=$transactionIndex, blockHash=$blockHashStr, blockNumber=$blockNumber, gasUsed=${gasUsed.toString()}, contractAddress=$contractAddressStr}")
+    val infoNonConsensusStr: String =
+      String.format(
+        s" - (receipt non consensus data) {txHash=$txHashStr, txIndex=$transactionIndex, blockHash=$blockHashStr, blockNumber=$blockNumber, gasUsed=${gasUsed
+            .toString()}, contractAddress=$contractAddressStr}"
+      )
 
-     consensusDataReceipt.toString.concat(infoNonConsensusStr)
+    consensusDataReceipt.toString.concat(infoNonConsensusStr)
   }
 
   override def equals(obj: Any): Boolean = {
     obj match {
       case other: EthereumReceipt =>
         consensusDataReceipt.equals(other.consensusDataReceipt) &&
-          new ByteArrayWrapper(transactionHash).equals(new ByteArrayWrapper(other.transactionHash)) &&
-          transactionIndex.equals(other.transactionIndex) &&
-          new ByteArrayWrapper(blockHash).equals(new ByteArrayWrapper(other.blockHash)) &&
-          blockNumber.equals(other.blockNumber) &&
-          gasUsed.equals(other.gasUsed) &&
-          new ByteArrayWrapper(contractAddress).equals(new ByteArrayWrapper(other.contractAddress))
+        util.Arrays.equals(
+          transactionHash,
+          0,
+          transactionHash.length - 1,
+          other.transactionHash,
+          0,
+          other.transactionHash.length - 1
+        ) &&
+        transactionIndex.equals(other.transactionIndex) &&
+        util.Arrays.equals(blockHash, 0, blockHash.length - 1, other.blockHash, 0, other.blockHash.length - 1) &&
+        blockNumber.equals(other.blockNumber) &&
+        gasUsed.equals(other.gasUsed) &&
+        util.Arrays.equals(
+          contractAddress.get,
+          0,
+          if (contractAddress.get.length > 0) contractAddress.get.length - 1 else 0,
+          other.contractAddress.get,
+          0,
+          if (other.contractAddress.get.length > 0) other.contractAddress.get.length - 1 else 0
+        )
 
       case _ => false
     }
   }
 
-  override def hashCode: Int =  {
+  override def hashCode: Int = {
     var result = consensusDataReceipt.hashCode()
     result = 31 * result + util.Arrays.hashCode(transactionHash)
     result = 31 * result + Integer.hashCode(transactionIndex)
     result = 31 * result + util.Arrays.hashCode(blockHash)
     result = 31 * result + Integer.hashCode(blockNumber)
     result = 31 * result + gasUsed.hashCode()
-    result = 31 * result + util.Arrays.hashCode(contractAddress)
+    result = 31 * result + util.Arrays.hashCode(contractAddress.get)
```

Comment:
In case `contractAddress == None` -> it will throw an exception.
same in other places.